herd/AArch64Sem.ml

```diff
@@ -1218,13 +1254,41 @@ module Make
       let simd_stp var addr1 rd1 rd2 ii =
         let open AArch64Base in
         let access_size = tr_simd_variant var in
-        ((read_reg_neon true rd1 ii >>= fun v1 ->
-        write_mem access_size aexp Access.VIR addr1 v1 ii)
-        >>|
-        (neon_sz_k var >>= fun os ->
-        M.add addr1 os >>|
-        read_reg_neon true rd2 ii >>= fun (addr2,v2) ->
-        write_mem access_size aexp Access.VIR addr2 v2 ii)) >>= B.next2T
+        if access_size == MachSize.S128 then
```

Comment:
> Can we factor out the code in simd_str and use it here? Not sure, if it would look much better but it looks like large parts of the code there and here are the same.

I wanted to make the same suggestion of sharing code. In particular, a function that splits a 128 bits read (resp. write) into two 64 bits reads (resp . write) may be called  by the implementations of load (resp. store) and load (resp. store) pair.

Reply:
Addressed that in the new commit I've pushed :)